### ISSUE TITLE
Assumes Hex View for .p and .pp file extensions.

### ISSUE THREAD

**Issue by 3FLLC** [FMXExpress/Trolledge#13]
Hello, I am evaluating your editor as a possible Windows/Mac front-end to our compiler. However, opening standard Apple .p for Pascal Source - jumps to Hex View ... is there a way to disable this?

It also does it for Free Pascal .pp Pascal Source.

Also, would you be open to adding support to Ctrl-R or F9, to run the current source or project through the command line shell in your configuration screen. I have code I can give for both platforms you support that should work in Delphi - to shell and trap output, either as a returned String w/ CRLF - to render as a stringlist, or to shell and trap via PIPEs.

Looking forward to working with you on these 2 topics.

**Comment by FMXExpress** [comment]
Yes, the .p and .pp file types could be added to https://github.com/FMXExpress/Trolledge/blob/master/uFileTypeHelper.pas

How compilation would work would be a command line setting could be defined and then a compile action would be setup to execute the command line against the current file.

Theoretically it would also be possible to compile with FMXLinux now which would give it Linux support as well.

**Comment by webbrain-one** [comment]
Hi, thanks for the report. I’ve opened PR #15 to fix this. It stops the editor from forcing `.p` and `.pp` files into Hex View and opens them as normal Pascal source text. Please review the linked PR: https://github.com/FMXExpress/Trolledge/pull/15
